src/search/videoId.ts

```diff
@@ -0,0 +1,54 @@
+import client from "../api/client.ts";
+import {
+  CHANNEL_ID,
+  MAX_RESULTS,
+  PART,
+  FIELDS,
+  TYPE,
+  VIDEO_DURATION,
+} from "../utils/constants.ts";
+import { ApiError, ParsingError } from "../errorHandle/errorTypes.ts";
+import { getErrorMessage } from "../errorHandle/errorMessage.ts";
+import { VideosIdSchema } from "../schema/VideosIdSchema.ts";
+
+async function getVideosId(): Promise<string[]> {
+  try {
```

Comment:
I wouldn't say I like having a nested try/catch, let's try to avoid this pattern in the future, however, it is ok for this specific example.
Please take a look at this [article](https://stackoverflow.com/questions/73533145/is-it-good-practice-to-use-try-inside-try-block-to-catch-different-errors-jav) you can learn when it is ok to follow this approach.